### ISSUE TITLE
iPhone iOS 9.02 & 9.3.1 both can't work

### ISSUE THREAD

**Issue by greenSnot** [mudcube/MIDI.js#178]
In official site,I only heard noise,but work well on my laptop.
I have tried mobile safari and mobile chrome(ver 50) which support WebAudio API.

**Comment by greenSnot** [comment]
iOS doesn't support autoplay for the "<audio>" element(should be triggered by touchstart or other events), MIDI.js generate so many <audio> that play different note(I guess) so cause this issue...I suggest to generate to one tag <audio> can solve this question.

**Comment by hmoffatt** [comment]
Is there a fix / workaround for this?

**Comment by hmoffatt** [comment]
There were some bugs in Safari on iOS 9.0-9.2. http://www.williammalone.com/posts/ios-9-web-audio-api-safari-mobile-issue/

I modified the examples to only init the audio context after the user presses a button. It seems maybe iOS Safari is trying to stop pages playing automatically. This sort of worked, in that the Basic one note example now works, but the MIDIPlayer example still stutters and makes noise after just a few notes. That might be what @greenSnot is describing.

**Comment by silverhawk184** [comment]
Is there a fix for this? It is a huge issue. I am willing to offer up a bounty to see it fixed.

**Comment by hmoffatt** [comment]
Which bit exactly @silverhawk184 ?

**Comment by silverhawk184** [comment]
If you test on an IOS device (iPhone 6s+ v9.3.3) (the pirate tuna example) all the notes turn into a jumbled mess of pink noise within a few seconds. Then if you press pause, it slowly becomes discernible again before it stops.

**Comment by hmoffatt** [comment]
Yes, the scheduling of all the WebAudio sources far in to the future (potentially 1000 nodes) is killing the iPad. That's a different issue that this one though which is that iOS Safari won't let you make any sound except in response to a user event.

**Comment by silverhawk184** [comment]
Correct, but as posts #1 references the "noise", it is the cause of the issue being filed. It was then followed up with the autoplay issue.
